pkgs/development/libraries/faad2/default.nix

```diff
@@ -12,8 +13,11 @@ stdenv.mkDerivation rec {
     sha256 = "1db37ydb6mxhshbayvirm5vz6j361bjim4nkpwjyhmy4ddfinmhl";
   };
 
+  patches = [ optional hdcSupport ./faad2-hdc-support.patch ];
```

Comment:
This is making an array of three things:

1. the `optional` function
2. the value of `hdcSupport`
3. the patch

To make it do what you expect, you'll need:

```
patches = [ (optional hdcSupport ./faad2-hdc-support.patch) ];
```

Reply:
Thanks for the hint